raiden/tests/unit/transfer/mediated_transfer/test_initiatorstate.py

```diff
@@ -1024,6 +1056,56 @@ def test_initiator_handle_contract_receive_secret_reveal():
     })
 
 
+def test_initiator_handle_contract_receive_emptyhash_secret_reveal():
+    """ Initiator must not accept contract receive secret reveal with emptyhash
+    """
+    amount = UNIT_TRANSFER_AMOUNT * 2
+
+    channel1 = factories.make_channel(
+        our_balance=amount,
+        token_address=UNIT_TOKEN_ADDRESS,
+        token_network_identifier=UNIT_TOKEN_NETWORK_ADDRESS,
+    )
+    pseudo_random_generator = random.Random()
+
+    channel_map = {
+        channel1.identifier: channel1,
+    }
+
+    available_routes = [
+        factories.route_from_channel(channel1),
+    ]
+
+    block_number = 10
+    current_state = make_initiator_manager_state(
+        routes=available_routes,
+        transfer_description=factories.UNIT_TRANSFER_DESCRIPTION,
+        channel_map=channel_map,
+        pseudo_random_generator=pseudo_random_generator,
+        block_number=block_number,
+    )
+
+    transfer = current_state.initiator.transfer
+
+    assert transfer.lock.secrethash in channel1.our_state.secrethashes_to_lockedlocks
+
+    state_change = ContractReceiveSecretReveal(
+        transaction_hash=factories.make_transaction_hash(),
+        secret_registry_address=factories.make_address(),
+        secrethash=transfer.lock.secrethash,
+        secret=EMPTY_HASH,
+        block_number=transfer.lock.expiration,
+    )
+
+    iteration = initiator_manager.handle_onchain_secretreveal(
+        payment_state=current_state,
+        state_change=state_change,
+        channelidentifiers_to_channels=channel_map,
+        pseudo_random_generator=pseudo_random_generator,
+    )
+    assert len(iteration.events) == 0
```

Comment:
Maybe also check that the lock hasn't been moved?
```
assert transfer.lock.secrethash in channel1.our_state.secrethashes_to_lockedlocks
```

Reply:
sure will do that.